src/CartItem.java

```diff
@@ -22,11 +22,21 @@ public int getQuantity() {
     }
 
     public void setQuantity(int quantity) {
+        if (quantity < 0) {
+            throw new IllegalArgumentException("Quantity cannot be negative.");
+        }
         this.quantity = quantity;
     }
 
-    public boolean equals(CartItem item) {
-        return this.itemName.equals(item.getName());
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) return true;
+        if (obj == null || getClass() != obj.getClass()) return false;
+
+        CartItem other = (CartItem) obj;
+        return itemName.equals(other.itemName) &&               // check name
+               Double.compare(price, other.price) == 0 &&       // check price
+               quantity == other.quantity;                      // check quantity
```

Comment:
This implementation is really cool!
Since we are also updating the equals method, the PR title and description needs to be updated to reflect that this change is also being made to the code.
I just created issue #26 for the `equals()` method related changes, so you can mention that the PR fixes issue #26 as well.
We should also remove the comments as well.

```suggestion
    @Override
    public boolean equals(Object obj) {
        if (this == obj) return true;
        if (obj == null || getClass() != obj.getClass()) return false;

        CartItem other = (CartItem) obj;
        return itemName.equals(other.itemName) &&         
               Double.compare(price, other.price) == 0 &&
               quantity == other.quantity;
```